lib/Controller/ImageController.php

```diff
@@ -74,6 +74,26 @@ public function __construct(string $appName,
 		$this->logger = $logger;
 	}
 
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @param int $textFileId
+	 * @param string $imagePath
+	 * @return DataResponse
+	 */
+	public function insertImageFile(int $textFileId, string $imagePath): DataResponse {
+		try {
+			$insertResult = $this->imageService->insertImageFile($textFileId, $imagePath, $this->userId);
+			if (isset($insertResult['error'])) {
+				return new DataResponse($insertResult, Http::STATUS_BAD_REQUEST);
+			} else {
+				return new DataResponse($insertResult);
+			}
+		} catch (Exception $e) {
+			return new DataResponse(['error' => 'File insertion error: ' . $e->getMessage()], Http::STATUS_BAD_REQUEST);
```

Comment:
Let's not expose the actual exception message and rather log it or use a [HintException](https://github.com/nextcloud/server/blob/b78f3a57d1d05fb5f81ec8361139b64a3e54d7b0/lib/public/HintException.php#L36) for known cases. Otherwise exception messages might leak internal data.